rust/main/agents/validator/src/submit.rs

```diff
@@ -127,6 +138,61 @@ impl ValidatorSubmitter {
             })
             .await;
 
+            if latest_checkpoint.index < latest_seen_checkpoint_index {
+                if let Some(block_height) = latest_checkpoint.block_height {
+                    if block_height >= latest_seen_checkpoint_block_height {
+                        tracing::error!(
+                            ?latest_checkpoint,
+                            ?latest_seen_checkpoint_index,
+                            ?latest_seen_checkpoint_block_height,
+                            "Latest checkpoint index is lower than previously seen, but has a block height equal or greater.");
+
+                        let checkpoint = self.checkpoint(&tree);
+                        Self::panic_with_reorg(
+                            &self.reorg_reporter,
+                            &self.reorg_period,
+                            &self.checkpoint_syncer,
+                            tree.root(),
+                            &latest_checkpoint,
+                            &checkpoint,
```

Comment:
Since source code from `panic_with_reorg` was used only after the latest checkpoint was ingested into the tree, if there won't not be any reorg, `latest_checkpoint` and `checkpoint` would be the same, and we are panicking since they are different.

Now we calculating `checkpoint = self.checkpoint(&tree)` while we have not ingested all the checkpoints up to the `latest_checkpoint`. Therefore, we cannot really report it in this way. Probably, we need to adopt `panic_with_reorg` to support this new feature.